gps_utils.py

```diff
@@ -0,0 +1,84 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Utilities for doing operations on GPS locations.
+"""
+
+import math
+
+# World Geodetic System defined constants
+_WGS84_EARTH_RADIUS = 6378137.0
+_WGS84_ECCENTRICITY = 0.0818191908426
+
+
+class Location():
+  """An object for a location in the form of a set of coordinates.
+
+  Attributes:
+    latitude: a float for the latitude of the location in Decimal Degrees
+    longitude: a float for the longitude of the location in Decimal Degrees
+    altitude: a float for the altitude of the location in meters
+    x: a float for the x coordinate of the location in ECEF format
+    y: a float for the y coordinate of the location in ECEF format
+    z: a float for the z coordinate of the location in ECEF format
+  """
+
+  def __init__(self, latitude, longitude, altitude=0):
+    self.latitude = latitude
+    self.longitude = longitude
+    self.altitude = altitude
```

Comment:
can google map API be used to provide the real altitude at the lat/long coordinate if user did not enter it?

Reply:
maps only gives lat/lon, so I just fed all of the points into Elevation API from maps that gives elevation based on lat/lon and used those